src/modules/Dropdown/Dropdown.js

```diff
@@ -1273,7 +1338,7 @@ export default class Dropdown extends Component {
       useKeyOnly(selection, 'selection'),
       useKeyOnly(simple, 'simple'),
       useKeyOnly(scrolling, 'scrolling'),
-      useKeyOnly(upward, 'upward'),
+      useKeyOnly(upward || shouldOpenUpward, 'upward'),
```

Comment:
When we have props that need to be controlled, we move to them to being `autoControlledProps`.  See Dropdown.js around line 346.

- add `upward` to `static autoControlledProps = []`
- add `defaultUpward` as a `propType` (since it is now stateful, we accept a default value for it)
- move all usages of `upward` from `this.props.upward` to `this.state.upward`
- to change the value of `upward` use `this.trySetState({ upward: <new value> })`
- remove `shouldOpenUpward` as it is no longer needed.

Auto controlled props automatically defer to the user's prop value, if any.  Also, `this.trySetState()` will not update the state value if the user is using that prop.